app/controllers/BasicAuthenticationFilter.scala

```diff
@@ -0,0 +1,144 @@
+    else unauthorizedResult
+
+  private def isAuthorized(requestHeader: RequestHeader) = {
+    lazy val authorizedByHeader =
+      requestHeader.headers.get(AUTHORIZATION).exists(expectedHeaderValues)
+
+    lazy val authorizedByCookie =
+      requestHeader.cookies.get(COOKIE_NAME).exists(_.value == cookieValue)
+
+    authorizedByHeader || authorizedByCookie
+  }
+
+  private def addCookie(result: Future[Result]) =
+    result.map(_.withCookies(cookie))
+
+  private lazy val configuration = configurationFactory
+
+  private lazy val unauthorizedResult =
+    Future successful Unauthorized.withHeaders(WWW_AUTHENTICATE -> realm)
+
+  private lazy val COOKIE_NAME = "play-basic-authentication-filter"
+
+  private lazy val cookie = Cookie(COOKIE_NAME, cookieValue)
+
+  private lazy val cookieValue =
+    Crypto.sign(configuration.username + configuration.passwords)
+
+  private lazy val expectedHeaderValues =
+    configuration.passwords.map { password =>
+      val combined = configuration.username + ":" + password
+      val credentials = Base64.encodeBase64String(combined.getBytes)
+      basic(credentials)
+    }
+
+  private def realm = basic(s"""realm=\"${configuration.realm}"""")
+
+  private def basic(content: String) = s"Basic $content"
+}
+
+object BasicAuthenticationFilter {
+  def apply() = new BasicAuthenticationFilter(
+    BasicAuthenticationFilterConfiguration.parse(
+      play.api.Play.current.configuration
+    )
+  )
+
+  def apply(configuration: => Configuration) = new BasicAuthenticationFilter(
+    BasicAuthenticationFilterConfiguration parse configuration
+  )
+}
+
+case class BasicAuthenticationFilterConfiguration(
+                                                   realm: String,
+                                                   enabled: Boolean,
+                                                   username: String,
+                                                   passwords: Set[String],
+                                                   excluded: Set[String])
+
+object BasicAuthenticationFilterConfiguration {
+
+  private val defaultRealm = "Application"
+  private def credentialsMissingRealm(realm: String) =
+    s"$realm: The username or password could not be found in the configuration."
+
+  def parse(configuration: Configuration) = {
+
+    val root = "basicAuthentication."
+    def boolean(key: String) = configuration.getBoolean(root + key)
+    def string(key: String) = configuration.getString(root + key)
+    def seq(key: String) =
+      Option(configuration.underlying getValue (root + key)).map { value =>
+        value.valueType match {
+          case ConfigValueType.LIST => value.unwrapped.asInstanceOf[java.util.List[String]].asScala
+          case ConfigValueType.STRING => Seq(value.unwrapped.asInstanceOf[String])
+          case _ => sys.error(s"Unexpected value at `${root + key}`, expected STRING or LIST")
+        }
+      }
+
+    val enabled = boolean("enabled").getOrElse(true)
```

Comment:
It's actually set to false in the config file, but I can default it to false here too